addons/mail/static/src/owl/components/activity/activity.js

```diff
@@ -0,0 +1,228 @@
+        ev.preventDefault();
+        const action = {
+            type: 'ir.actions.act_window',
+            name: this.env._t("Schedule Activity"),
+            res_model: 'mail.activity',
+            view_mode: 'form',
+            views: [[false, 'form']],
+            target: 'new',
+            context: {
+                default_res_id: this.storeProps.activity.resId,
+                default_res_model: this.storeProps.activity.model,
+            },
+            res_id: this.storeProps.activity.id,
+        };
+        return this.env.do_action(action, {
+            on_close: () => {
+                this.storeDispatch('updateActivity', this.props.activityLocalId);
+                this.storeDispatch('refreshChatter', this.storeProps.activity.chatterLocalId);
+            }
+        });
+    }
+
+    _onClickPreviewMailTemplate(ev) {
+        ev.stopPropagation();
+        ev.preventDefault();
+        const templateId = ev.target.dataset.mailTemplateId;
+        const action = {
+            name: this.env._t("Compose Email"),
+            type: 'ir.actions.act_window',
+            res_model: 'mail.compose.message',
+            views: [[false, 'form']],
+            target: 'new',
+            context: {
+                default_res_id: this.storeProps.activity.resId,
+                default_model: this.storeProps.activity.model,
+                default_use_template: true,
+                default_template_id: Number(templateId),
+                force_email: true,
+            },
+        };
+        this.env.do_action(action, {
+            on_close: () => {
+                if (this.storeProps.activity.chatterLocalId) {
+                    this.storeDispatch('refreshChatter', this.storeProps.activity.chatterLocalId);
+                }
+            }
+        });
+    }
+
+    async _onClickSendMailTemplate(ev) {
+        ev.stopPropagation();
+        ev.preventDefault();
+        const templateId = ev.target.dataset.mailTemplateId;
+        await this.env.rpc({
+            model: this.storeProps.activity.model,
+            method: 'activity_send_mail',
+            args: [[this.storeProps.activity.resId], Number(templateId)],
+        });
+        if (this.storeProps.activity.chatterLocalId) {
+            this.storeDispatch('refreshChatter', this.storeProps.activity.chatterLocalId);
+        }
+    }
+
+    _onClickUploadDocument(ev) {
+        this._fileUploaderRef.comp.openBrowserFileUploader();
+    }
+
+    _onAttachmentCreated(ev) {
+        const { attachmentLocalId } = ev.detail;
+        this.storeDispatch('markActivityAsDone', this.props.activityLocalId, {
+            attachmentLocalIds: [attachmentLocalId],
+        });
+    }
+
+    _toggleDetailsVisibility() {
+        this.state.isDetailsVisible = !this.state.isDetailsVisible;
+    }
+
+}
+
+Activity.components = { ActivityMarkDoneButton, FileUploader };
+
+Activity.props = {
+    activityLocalId: {
+        type: String,
+    },
```

Comment:
What is the convention ? Because again, I can see the two way of writing it in many components ... (And I think I remember that seb or you said that 'type:X' was more easy to understand but I'm not sure of that)

Like here :

```
ChatWindowHeader.props = {
    chatWindowLocalId: String,
    hasCloseAsBackButton: {
        type: Boolean,
    },
    hasShiftLeft: {
        type: Boolean,
    },
    hasShiftRight: {
        type: Boolean,
    },
    isExpandable: {
        type: Boolean,
    },
};
```